liquibase-core/src/main/java/liquibase/command/CommandScope.java

```diff
@@ -134,6 +137,46 @@ public CommandResults execute() throws CommandExecutionException {
             provider.validate(this);
         }
 
+        // if the command has some arguments which are interactively promptable and which do not have values provided for them, ignore the defaults for those parameters and go into a prompting session
+        boolean initProjectModeOn = true;
+
+        if (initProjectModeOn) {
+            // Determine if the command has ANY parameters which are interactively promptable.
+            boolean commandHasAnyInteractivelyPromptableParameters = commandDefinition.getArguments().values().stream().anyMatch(CommandArgumentDefinition::isInteractivelyPromptable);
+            if (commandHasAnyInteractivelyPromptableParameters) {
+                // If the command does have any interactively promptable parameters, then generate a list of parameters which
+                // are interactively promptable and for which the user did not explicitly provide a value.
+//                SortedSet<ConfigurationDefinition<?>> registeredDefinitions = LiquibaseConfiguration.getInstance().getRegisteredDefinitions(true);
+                List<CommandArgumentDefinition<?>> interactivelyPromptableParametersWithDefaultValue =
+                        commandDefinition.getArgumentsInInsertionOrder().values().stream().filter(cad -> (getConfiguredValue(cad).wasDefaultValueUsed() || !getConfiguredValue(cad).found()) && cad.isInteractivelyPromptable())
+                                .collect(Collectors.toList());
+
+                if (!interactivelyPromptableParametersWithDefaultValue.isEmpty()) {
```

Comment:
Asking the user in this generic code about setting up a new liquibase.properties file and sample changelog seems too "init" specific. I think there are good use cases for prompting the user for arguments where we are never going to persisting those anywhere like for the password etc.

This logic of asking about saving values should belong in the init logic, not here.

Reply:
I intentionally put this logic here to support the future vision. In the future, running commands like `liquibase update` and omitting the `changeLogFile` parameter will automatically route the user into an interactive "init project" prompting session.